app/assets/javascripts/decidim/decidim_awesome/map/map.js

```diff
@@ -0,0 +1,177 @@
+
+    const svg = `<svg xmlns="http://www.w3.org/2000/svg" version="1.1" class="${className}" style="${style}">${path}${circle}${text}</svg>`;
+
+    return svg;
+  }
+});
+
+const popupMeetingTemplateId = "marker-meeting-popup";
+$meetingTmpl = $.template(popupMeetingTemplateId, $(`#${popupMeetingTemplateId}`).html());
+
+const popupProposalTemplateId = "marker-proposal-popup";
+$proposalTmpl = $.template(popupProposalTemplateId, $(`#${popupProposalTemplateId}`).html());
+
+let markerClusters = L.markerClusterGroup();
+
+function printQuery(data, map) {
+  var participatory_process = data.participatoryProcess;
+  var markerData = [];
+
+  if (participatory_process.hasOwnProperty("components")) {
+    participatory_process.components.forEach(element => {
+ 
+
+      if (element.__typename == "Meetings") {
+        element.meetings.edges.forEach(edge => {
+
+          var marker = L.marker([edge.node.coordinates.latitude, edge.node.coordinates.longitude], {
+            icon: new L.DivIcon.SVGIcon.DecidimIcon()
+          });
+          let node = document.createElement("div");
+          edge.node.link = '/processes/' + participatory_process.slug + '/f/' + element.id + '/meetings/' + edge.node.id;
+
+          $.tmpl($(`#${popupMeetingTemplateId}`), edge.node).appendTo(node);
+
+          marker.bindPopup(node, {
+            maxwidth: 640,
+            minWidth: 500,
+            keepInView: true,
+            className: "map-info"
+          }).openPopup();
+      
+          marker.addTo(map);
+          markerClusters.addLayer(marker);
+        });
+      } else if (element.__typename == "Proposals") {
+      
+        element.proposals.edges.forEach(edge => {
+          
+          var marker = L.marker([edge.node.coordinates.latitude, edge.node.coordinates.longitude], {
+            icon: new L.DivIcon.SVGIcon.DecidimIcon()
+          });
+          var node = document.createElement("div");
+          edge.node.link = '/processes/' + participatory_process.slug + '/f/' + element.id + '/proposals/' + edge.node.id;
+          $.tmpl($(`#${popupProposalTemplateId}`), edge.node).appendTo(node);
+      
+          marker.bindPopup(node, {
+            maxwidth: 640,
+            minWidth: 500,
+            keepInView: true,
+            className: "map-info"
+          }).openPopup();
+      
+          markerClusters.addLayer(marker);
+        });
+      }
+    });
+  }
+
+    map.addLayer(markerClusters);
+    // map.fitBounds(bounds, { padding: [100, 100] });
+}
+
+$(() => {
+  const mapId = "awesome-map";
+  const $map = $(`#${mapId}`);
+  const participatory_space = $map.data('participatory_space');
+
+  var graphql_query = 'query ($participatory_space: String!, $lang: String!)\
+    {\
+      participatoryProcess(slug: $participatory_space) {\
+        id\
+        slug\
+        components(filter: {withGeolocationEnabled: true}) {\
+          id\
+          __typename\
+          ... on Meetings {\
+            meetings {\
```

Comment:
```suggestion
            meetings(after: $meetings_cursor) {\
                pageInfo {
                    endCursor
                    hasNextPage
              }
```
We will need to paginate and extract all the results while draw the points progressively